components/src/core/Drawer/Drawer.tsx

```diff
@@ -110,6 +110,7 @@ const Root = styled(MUIDrawer, {
         position: 'relative',
         boxShadow: sideBorder ? 'none' : theme.shadows[4],
         borderWidth: sideBorder ? 1 : 0,
+        zIndex: 'auto',
```

Comment:
If the solution was already working on dev before, updating the component library styles should not be required.   The fix should be scoped to the demo application, not our component library.